examples/overhead_bench.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+import logging
+import time
+
+import torch
+from llama import Llama
+
+from flex_model.core import FlexModel, HookFunction
+from flex_model.utils import setup_logger
+
+
+logger = logging.getLogger(__name__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--log-level", type=str, default="info")
+    parser.add_argument(
+        "--checkpoint-dir", type=str, default="/model-weights/Llama-2-13b"
+    )
+    parser.add_argument(
+        "--tokenizer-dir",
+        type=str,
+        default="/model-weights/Llama-2-13b/tokenizer.model",
+    )
+    parser.add_argument("--max-seq-len", type=int, default=1024)
+    parser.add_argument("--max-batch-size", type=int, default=4)
+    parser.add_argument("--steps", type=int, default=100)
+
+    args = parser.parse_args()
+    return args
+
+
+def main(args):
+    setup_logger(args.log_level)
+    # Build the llama-2 model to benchmark on.
+    generator = Llama.build(
+        ckpt_dir=args.checkpoint_dir,
+        tokenizer_path=args.tokenizer_dir,
+        max_seq_len=args.max_seq_len,
+        max_batch_size=args.max_batch_size,
+    )
+    model = generator.model
+
+    # Synthetic batch.
+    data = torch.randint(0, 32000, (args.max_batch_size, args.max_seq_len)).cuda()
+
+    # Benchmark regular inference time without flex model.
+    start_t = time.time()
+    for i in range(args.steps):
+        _ = model(data, start_pos=0)
+    torch.cuda.synchronize()
+    without_t = (time.time() - start_t) / args.steps
+
+    # Benchmark inference time with flex model, one hook per layer.
+    out_dict = {}
+    flex_model = FlexModel(
+        model,
+        out_dict,
+        tensor_parallel_size=torch.distributed.get_world_size(),
+    )
+    hidden_dim = model.layers[0].feed_forward.w1.out_features
+    n_layers = model.params.n_layers
+    # TODO (mchoi): This is a common pattern, possible to automatically do this
```

Comment:
Given that this is just to spin up a simple test, we can probably leave these customizations out imo. MLP activations should already give an idea of the worst-case runtimes for activation caching.

Reply:
These aren't customizations, this is how we grep out the hidden dim and number of layers from the llama model.